lib/open_project/scm/adapters/git.rb

```diff
@@ -91,6 +114,82 @@ def check_availability!
           raise Exceptions::ScmUnavailable
         end
 
+        def checkout_repository!
+          Rails.logger.info "Checking out #{checkout_uri} to #{checkout_path}"
+
+          FileUtils.mkdir_p checkout_path.parent
+          %x(git clone #{checkout_uri} #{checkout_path})
```

Comment:
Don't use `%x`, it spawns a shell and is susceptible to all sorts of attacks when you pass it user input. Use the `capture_out` or `popen3` helpers in the  `LocalClient` adapter superclass.